lib/Mojo/Promise.pm

```diff
@@ -214,8 +214,19 @@ sub _settle {
     $self = $self->new;
   }
 
+  !$self->{_semi_settled}++ or return $self;
+
   if ($thenable && $status eq 'resolve') {
-    $results[0]->then(sub { $self->resolve(@_); () }, sub { $self->reject(@_); () });
+    if (refaddr $results[0] == refaddr $self) {
+      undef $self->{_semi_settled};
+      $self->reject('Chaining cycle detected');
```

Comment:
Isn't the JavaScript exception `Chaining cycle detected for promise`?

Reply:
Addressed.

**node.js** exception is `Chaining cycle detected for promise [Promise]`. Remembering how **node.js** outputs `[object Object]` when you stringify an object (as in `console.log('hi: ' + {})`), it came to my mind that it would be more correct, and helpful to developers who are investigating their own bugs at the same time, if I stringified $self (a promise) in this message too (as in: `"Chaining cycle detected for promise $self"`).

If you don't agree with this I will remove $self from that string.